src/mon/FSCommands.cc

```diff
@@ -255,12 +239,15 @@ class FsNewHandler : public FileSystemCommandHandler
     pg_pool_t const *metadata_pool = mon->osdmon()->osdmap.get_pg_pool(metadata);
     ceph_assert(metadata_pool != NULL);  // Checked it existed above
 
-    int r = _check_pool(mon->osdmon()->osdmap, data, POOL_DATA_DEFAULT, force, &ss);
+    bool allow_overlay = false;
+    cmd_getval(cmdmap, "allow_dangerous_metadata_overlay", allow_overlay);
+
+    int r = _check_pool(mon->osdmon()->osdmap, data, POOL_DATA_DEFAULT, force, allow_overlay, &ss);
```

Comment:
since the config option being checked is `allow_dangerous_metadata_overlay` is it correct to allow data pool (`POOL_DATA_DEFAULT`) to be used for data overlay ?

Reply:
Keeping behaviour as earlier for now.
Please let me know if you want me to change it.